src/app/todos/todo.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+
+import { Todo, TodoStatus } from './todo.model';
+
+@Injectable()
+export class TodoService {
+  todosChanged = new Subject<Todo[]>();
+
+  private todos: Todo[] = [
+    new Todo( '1', 'Install Angular', TodoStatus.Done),
+    new Todo( '2', 'Add Edit TODO title', TodoStatus.Done),
+    new Todo( '3', 'Create basic TODO app', TodoStatus.InProgress),
+    new Todo( '4', 'Change UI from Bootstrap to Angular material', TodoStatus.Backlog),
+    new Todo( '5', 'Make UI like microsoft TODO app', TodoStatus.Backlog),
+    new Todo( '6', 'Add Edit TODO title', TodoStatus.Backlog),
+    new Todo( '7', 'Add Firebase API', TodoStatus.Backlog),
+    new Todo( '8', 'Add Drag-n-drop status columns', TodoStatus.Backlog),
+  ];
+
+  constructor() {}
+
+  // @ts-ignore
+  getTodo(id: string): Todo {
+    const todo = this.todos.find(todo => todo.id === id);
+
+    if (todo) {
+      return todo;
+    }
+  }
+
+  getTodos(): Todo[] {
+    return [...this.todos];
```

Comment:
Just wondering, why would you need to destructure and create a new array? What is this method for?)

Reply:
I use it to avoid unintended side effects that can occur when passing the original array directly

Reply:
You can wrap the array in an observable using the rxjs operator, and subscribe to it in the component and get the original array and process it there as it is necessary specifically for this component. Feel free to use RxJs